### ISSUE TITLE
Feature: ALSA fallback device

### ISSUE THREAD

**Issue by rid9** [mpv-player/mpv#6059]
I haven't seen this option, and I think it would be really nice to have, if possible.

As far as I can tell, you can specify with "auto" that mpv should look for the first working ALSA device in a predefined list that cannot be changed. What I would like is to have a command line option to specify the list of ALSA devices to try (maybe something like `--audio-device=alsa/bluealsa,alsa,...`). If the first device doesn't work, then use the second, etc.

It would be even nicer if this could happen dynamically. For example, if I'm currently listening with device `alsa/bluealsa` and the device goes away, then pause the video, go through all the devices again, find the first working one, use that, then resume the video.

**Comment by keithbowes** [comment]
Yeah, this is the one missing feature in moving from --ao to --audio-device. Why mpv couldn't have just kept mplayer's way of specifying the device through --ao is anyone's guess. Though, yes, `audio-device=alsa/dmixer` probably always works (assuming you've set up a dmix device in your ALSA config) and the equivalent in my mplayer config (I don't know why I keep mplayer around), `ao=alsa:device=dmixer,pulse,jack:autostart,` is probably redundant, as it's doubtful that PulseAudio or JACK will ever be reached.
